File: GotadeOuroDefault.Web/Modules/ServerTypes/Servico/TipodeServicoRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface TipodeServicoRow {
    IdTipoServico?: number;
    Nome?: string;
}

export abstract class TipodeServicoRow {
    static readonly idProperty = 'IdTipoServico';
    static readonly nameProperty = 'Nome';
    static readonly localTextPrefix = 'Servico.TipodeServico';
    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<TipodeServicoRow>();
}

File: GotadeOuroDefault.Web/Modules/Servico/TipodeServico/TipodeServicoRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace GotadeOuroDefault.Servico;

[ConnectionKey("Default"), Module("Servico"), TableName("TipoServico")]
[DisplayName("Tipode Servico"), InstanceName("Tipode Servico")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
public sealed class TipodeServicoRow : Row<TipodeServicoRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id Tipo Servico"), Column("idTipoServico"), PrimaryKey, NotNull, IdProperty]
    public int? IdTipoServico
    {
        get => fields.IdTipoServico[this];
        set => fields.IdTipoServico[this] = value;
    }

    [DisplayName("Nome"), Size(50), QuickSearch, NameProperty]
    public string Nome
    {
        get => fields.Nome[this];
        set => fields.Nome[this] = value;
    }

    public class RowFields : RowFieldsBase
    {
        public Int32Field IdTipoServico;
        public StringField Nome;

    }
}